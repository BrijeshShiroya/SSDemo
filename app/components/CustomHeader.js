import PropTypes from 'prop-types';
import React from 'react';
import { Image, Text, TouchableOpacity } from 'react-native';
import { SafeAreaView } from 'react-navigation';
import styles from './styles/CustomHeaderStyles';

const leftSection = (leftIcon, leftIconStyle, leftOnPress) => (
  <TouchableOpacity
    activeOpacity={0.7}
    hitSlop={styles.hitSlop}
    onPress={leftOnPress}
  >
    <Image source={leftIcon} style={[styles.iconStyle, leftIconStyle]} />
  </TouchableOpacity>
);
const CustomHeader = ({
  title = 'Header',
  titleStyle,
  headerStyle,
  left,
  leftIcon,
  leftIconStyle,
  leftOnPress
}) => (
  <SafeAreaView
    style={[styles.headerContainer, headerStyle]}
    forceInset={{ bottom: 'never' }}
  >
    {left && leftSection(leftIcon, leftIconStyle, leftOnPress)}
    <Text style={[styles.title, titleStyle]}>{title}</Text>
  </SafeAreaView>
);

CustomHeader.propTypes = {
  left: PropTypes.bool,
  leftIcon: PropTypes.object,
  leftOnPress: PropTypes.func,
  leftIconStyle: PropTypes.object,
  title: PropTypes.string,
  titleStyle: PropTypes.object,
  headerStyle: PropTypes.object
};

export default CustomHeader;
