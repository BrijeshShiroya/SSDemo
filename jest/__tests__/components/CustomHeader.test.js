import 'react-native';
import React from 'react';
import CustomHeader from '../../../app/components/CustomHeader';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
it('CustomHeader renders correctly', () => {
  jest.useFakeTimers();
  const customHeaderComponent = renderer.create(<CustomHeader />).toJSON();
  expect(customHeaderComponent).toMatchSnapshot();
});
