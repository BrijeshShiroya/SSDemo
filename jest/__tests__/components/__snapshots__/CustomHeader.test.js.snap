// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomHeader renders correctly 1`] = `
<View
  forwardedRef={[Function]}
  onLayout={[Function]}
  pointerEvents="box-none"
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgb(0, 0, 0)",
      "flexDirection": "row",
      "height": 118.57142857142857,
      "justifyContent": "space-between",
      "paddingBottom": 0,
      "paddingLeft": 24,
      "paddingRight": 24,
      "paddingTop": 20,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "white",
          "fontFamily": "AvenirNext-Bold",
          "fontSize": 29.57142857142857,
          "textAlign": "center",
        },
        undefined,
      ]
    }
  >
    Header
  </Text>
</View>
`;
